File: data/src/main/java/org/cryptomator/data/cloud/okhttplogging/HeaderNames.java
package org.cryptomator.data.cloud.okhttplogging;

import java.util.HashSet;
import java.util.Set;

class HeaderNames {

	private final Set<String> lowercaseNames = new HashSet<>();

	public HeaderNames(String... headerNames) {
		for (String headerName : headerNames) {
			lowercaseNames.add(headerName.toLowerCase());
		}
	}

	public boolean contains(String headerName) {
		return lowercaseNames.contains(headerName.toLowerCase());
	}

}

File: data/src/main/java/org/cryptomator/data/cloud/okhttplogging/HeaderNames.kt
package org.cryptomator.data.cloud.okhttplogging

import java.util.HashSet

internal class HeaderNames(vararg headerNames: String) {

	private val lowercaseNames: MutableSet<String> = HashSet()

	operator fun contains(headerName: String): Boolean {
		return lowercaseNames.contains(headerName.lowercase())
	}

	init {
		headerNames.mapTo(lowercaseNames) { it.lowercase() }
	}
}
